### PR TITLE
:sparkles: implement full list view and fetch more

### PR DESCRIPTION
**Pull request by NekoLyn**
a simple full list view, need to improve styles and implement button functions, can raise a follow up ticket according to feedback

TODO:
1. need pin list? if need, should put in higher level so pin list can be remembered after switching view mode or pages
3. bug: default fetch page size is 11 when changing sort options in full list view

image

image

### CODE REVIEW

File: src/pages/search-page/SearchPage.tsx [hunk 1]
```diff
@@ -427,35 +427,35 @@ const SearchPage = () => {
         }}
         gap={2}
       >
-        {/* Ignore the FULL_LIST view for now, wait for future design */}
-        {selectedLayout !== SearchResultLayoutEnum.FULL_MAP && (
-          <Box>
-            <ResultSection
-              onClickCard={onClickCard}
+        <Box>
```
Comment (utas-raymondng) [inline src/pages/search-page/SearchPage.tsx:430]: I prefer follow the same patter then the ResultSection have an attribute like below showFullMap, where you can showFullList

File: src/components/result/FullListAndDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      const paramPaged = createSearchParamFrom(componentParam, {
+        pagesize: FULL_LIST_PAGE_SIZE,
+      });
+      dispatch(fetchResultWithStore(paramPaged));
+    }
+    //only for fetching initial data
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const fetchMore = useCallback(async () => {
```
Comment (utas-raymondng) [inline src/components/result/FullListAndDetails.tsx:48]: I think the fetch is done on the top level so you do not need to fetch it here?

**Review by utas-raymondng — CHANGES_REQUESTED** [review]
Lets have a quick chat on some of these changes

File: src/components/result/FullListAndDetails.tsx (hunk 2 above)
Comment (NekoLyn) [reply src/components/result/FullListAndDetails.tsx]: thanks! I created a useFetchData hook for reusing in different components

File: src/pages/search-page/SearchPage.tsx (hunk 1 above)
Comment (NekoLyn) [reply src/pages/search-page/SearchPage.tsx:430]: I modified to pass showFullList and showFullMap

**Review by utas-raymondng — APPROVED** [review]
LGTM

**Pull request merged**